riscv-syscalls/src/lib.rs

```diff
@@ -26,15 +36,15 @@ macro_rules! syscalls {
             }
         }
 
-        impl From<Syscall> for u32 {
+        impl From<Syscall> for u8 {
             fn from(syscall: Syscall) -> Self {
                 syscall as Self
             }
         }
```

Comment:
This looks a bit dangerous. I think this should be removed, and if you need to convert number to enum, use the TryFrom.